presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/DriverConnectionFactory.java

```diff
@@ -56,6 +57,8 @@ public DriverConnectionFactory(Driver driver, String connectionUrl, Properties c
     public Connection openConnection()
             throws SQLException
     {
-        return driver.connect(connectionUrl, connectionProperties);
+        Connection connection = driver.connect(connectionUrl, connectionProperties);
+        checkState(connection != null, "Driver returned null connection");
```

Comment:
maybe you could log `connectionUrl` and `connectionProperties`

Reply:
Yes, i wanted to. But connectionProperties usually contain credentials, so I would need to remove the from output. Also, connection url also can have credentials encoded in it, so I decided it's better not to print them.